tests/tests/tier0/bluechi-is-online-agent-monitor/test_bluechi_is_online_agent_monitor.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright Contributors to the Eclipse BlueChi project
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+import logging
+import threading
+import time
+from typing import Dict
+
+from bluechi_test.config import BluechiAgentConfig, BluechiControllerConfig
+from bluechi_test.machine import BluechiAgentMachine, BluechiControllerMachine
+from bluechi_test.test import BluechiTest
+from bluechi_test.util import Timeout, get_test_env_value_int
+
+LOGGER = logging.getLogger(__name__)
+
+NODE_FOO = "node-foo"
+SLEEP_DURATION = get_test_env_value_int("SLEEP_DURATION", 2)
+
+
+class MonitorResult:
+    def __init__(self):
+        self.result = None
+        self.output = ""
+
+
+def monitor_command(node: BluechiAgentMachine, monitor_result: MonitorResult):
+    monitor_result.result, monitor_result.output = (
+        node.bluechi_is_online.monitor_agent()
+    )
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+    node_foo = nodes[NODE_FOO]
+
+    # Test 1: Agent is running, no monitor output expected
+    LOGGER.debug("Starting bluechi-agent.")
+    monitor_result_test_one = MonitorResult()
+    monitor_thread = threading.Thread(
+        target=monitor_command, args=(node_foo, monitor_result_test_one)
+    )
+
+    monitor_thread.start()
+    try:
+        with Timeout(SLEEP_DURATION, "Timeout while monitoring agent"):
+            monitor_thread.join()
+    except TimeoutError:
+        LOGGER.debug("Timeout reached while monitoring agent. Attempting to terminate.")
+
+    assert (
+        monitor_result_test_one.result is None
+    ), "Monitor command should not produce output when agent is running."
+
+    # Test 2: Stop bluechi-agent and verify monitoring detects the failure
+    LOGGER.debug("Starting monitor thread before stopping the agent.")
+    monitor_result_test_two = MonitorResult()
+    monitor_thread = threading.Thread(
+        target=monitor_command, args=(node_foo, monitor_result_test_two)
+    )
+    monitor_thread.start()
+    time.sleep(SLEEP_DURATION)
+
+    LOGGER.debug("Stopping the agent.")
+    node_foo.systemctl.stop_unit("bluechi-agent")
+
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+    monitor_thread.join()
+    assert (
+        monitor_result_test_two.result is not None
+        and monitor_result_test_two.output != ""
+    ), "Monitor command should produce an output when agent is stopped."
```

Comment:
NIT: I think we could check the `.result` more specifically:
```suggestion
    assert (
        monitor_result_test_two.result == 1
        and monitor_result_test_two.output != ""
    ), "Monitor command should produce an output when agent is stopped."
```

Reply:
Using 'monitor_result_test_two.result == 1' failed, Based on the test logs, monitor_result_test_two.result is currently captured as False when the agent is offline, rather than the exit code (1).
Also, when I run this test manually the command outputs text and not 1:
~$ sudo bluechi-is-online agent --monitor
org.eclipse.bluechi.Agent is offline
~$ echo $?
1

To make monitor_result_test_two.result == 1 work, we’d need to modify the monitor_agent() function to explicitly return the exit code from the bluechi-is-online command
or use 'monitor_result_test_two.result is False'
but that approach wouldn't align with the other bluechi-is-online tests.
Please Let me know your preference!

Reply:
Ah, true. I forgot that we already map the return code to a boolean. In that case, lets keep it as it is and not check for `False` - in these monitor tests its semantically the same.